Hamocc hybrid coord2

TomasTorsvik opened the pull request:
Suggestion for change in iHAMOCC bulk mixed layer index "kmle" from parameter value 2 to value set by OBL depth.

TomasTorsvik committed:
Change mixed layer depth index "kmle" from parameter to 2D field.

TomasTorsvik committed:
Set "kmle" according to index "kOBL" of level above OBL depth.

JorgSchwinger commented on hamocc/mo_vgrid.F90:
```diff
@@ -114,6 +115,7 @@ subroutine set_vgrid(kpie,kpje,kpke,pddpo)
 !$OMP END PARALLEL DO
 
 
+  kmle(:,:) =2
```
This will overwrite the values set in blom2hamocc every time step.

TomasTorsvik replied:
Thanks! That was left over from the first test implementation. It should not be used when importing the kOBL field from BLOM.

JorgSchwinger replied:
It might be useful to have the possibility though to set kmle=2 for backwards-compatibility(?). Because the cvmix-routine will not (necessarily) return the index 2 if the old vertical layer structure is activated.

TomasTorsvik replied:
I Agree. I can set this in blom2hamocc with a dependency on the value of `vcoord_type_tag`.

TomasTorsvik committed:
Don't reset kmle for each call to set_vgrid.

TomasTorsvik commented:
Perhaps there should be some sanity check for `kmle`? Is is best to do this in `blom2hamocc` when copying data from `kOBL`, or in subroutines where `kmle` is actually used?

JorgSchwinger commented:
I think if a sanity check would be needed, this should be done in BLOM (and I guess that cvmix already takes care of this).

TomasTorsvik committed:
Use static kmle value for isopycninc coordinates.

TomasTorsvik commented:
This PR may also be postponed until after the hybrid coordinate framework has been merged into the master branch.

matsbn commented:
There seems to be some conflicts after I merged #170 into feature-hybrid_coord2.

TomasTorsvik commented:
This branch needs the fixes from #172 . I could cherry-pick them into this branch, but that will become messy when we want to merge with `master` later on. I think the cleaner way is to update `master` with #172, merge `master` into `feature-hybrid_coord2`, and then rebase the `hamocc-hybrid_coord2` branch on top of the updated `feature-hybrid_coord2` branch.

TomasTorsvik committed:
bugfix: Add (i,j) indices to kOBL

TomasTorsvik commented:
@matsbn 
I have updated the `hamocc-hybrid_coord2` branch with the latest changes in `master` and rebased my local branch on top of `feature-hybrid_coord2` branch. This should eliminate conflicts in the code. The code compiles with meson with enabled ecosys flag.
